components/datetime/src/neo.rs

```diff
@@ -562,8 +559,28 @@ where
         P: ?Sized + AllAnyCalendarFormattingDataMarkers<FSet>,
         L: DecimalFormatterLoader + AnyCalendarLoader,
     {
-        let calendar = AnyCalendarLoader::load(loader, (&prefs).into())
-            .map_err(DateTimeFormatterLoadError::Data)?;
+        // This will eventually need fallback data from the provider
+        let algo = prefs.calendar_algorithm.unwrap_or_else(|| {
+            let lang = prefs.locale_preferences.language();
+            if lang == language!("th") {
+                CalendarAlgorithm::Buddhist
+            } else if lang == language!("sa") {
+                CalendarAlgorithm::Hijri(Some(HijriCalendarAlgorithm::Umalqura))
+            } else if lang == language!("af") || lang == language!("ir") {
+                CalendarAlgorithm::Persian
+            } else {
+                CalendarAlgorithm::Gregory
+            }
+        });
```

Comment:
Wait, why is this logic moving into the datetime crate?

Reply:
Because the datetime crate needs to check that the resolved `CalendarAlgorithm` is supported